### ISSUE TITLE
[Bug / 01.00.10.05 / Win ] Forced closure simulation

### ISSUE THREAD

**Issue by MarcoPiazza83** [bambulab/BambuStudio#28]
Simulating a forced shutdown, on restart you are asked whether to load the last save. 

<img width="574" alt="Schermata 2022-07-18 alle 11 28 39" src="https://user-images.githubusercontent.com/4783556/179484123-654af48d-41cc-4dd6-80d0-469b0e99f333.png">

The loaded stl file loses triangles in the mesh.

Original

<img width="574" alt="Schermata 2022-07-18 alle 11 33 59" src="https://user-images.githubusercontent.com/4783556/179484167-fc8ee03a-972e-453e-b90f-be5a9879cbe4.png">

After forced shutdown

<img width="574" alt="Schermata 2022-07-18 alle 11 31 37" src="https://user-images.githubusercontent.com/4783556/179484201-b3c751de-59d2-4811-ab10-9dbedbe211c7.png">

**Comment by RobinWhitfield** [comment]
Where the numbers exist on those parts on the right, do they actually have depth? If so, how much depth is present?

Testing with the following object in both STEP, STL, and multiples of the STL I can't get it to lose any triangles.

![image](https://user-images.githubusercontent.com/63362209/179485427-fd094099-6fa3-4994-884e-45aedc8f48eb.png)

![image](https://user-images.githubusercontent.com/63362209/179485483-cd6d78f0-1b98-41a9-ab8f-b457a18cbae7.png)

![image](https://user-images.githubusercontent.com/63362209/179485639-48dd5be3-4628-4cf3-aed3-64424c4aa853.png)

![image](https://user-images.githubusercontent.com/63362209/179485762-724915b5-fb03-443d-83a3-b07a887ea217.png)

![image](https://user-images.githubusercontent.com/63362209/179486210-e87625b8-d43b-4cde-86ce-7adebf348d18.png)

![image](https://user-images.githubusercontent.com/63362209/179486356-e8007626-fcbd-40af-9f86-42427b023e12.png)

**Comment by MarcoPiazza83** [comment]
@RobinWhitfield 

Right click-> reduce triangles.
Same.

did you force close from task manager?

**Comment by RobinWhitfield** [comment]
Yeah if I right click and reduce triangles then I get exactly the same triangle count as the box bottom right says.

I was force closing with task manager, and getting the same unsaved project message.

I'm presuming it's something specific with your model, as it's the numbers that are disappearing, and they look to have either a very small depth, or no depth at all.

If you slice the original file, does it do anything with the numbers, or is it trying to ignore them thinking they're not relevant?

**Comment by MarcoPiazza83** [comment]
@RobinWhitfield 

Could you please share your stl file?
So I perform more tests.
I had the same error with other stl as well.
The file in screenshot is a thingiverse download.

**Comment by RobinWhitfield** [comment]
It's just a random file I generated for testing.

I've just tested it with a load of Benchy's grouped as an assembly and it's seeming OK.
The exact model being the .STL from here: https://www.thingiverse.com/thing:763622/files

Before
![image](https://user-images.githubusercontent.com/63362209/180228405-d3ba1392-b9ac-4531-b45a-8e5bed9bdfc5.png)

After
![image](https://user-images.githubusercontent.com/63362209/180228477-f64d00f5-8b81-4086-bd6e-6bcce46dde46.png)



If you want to share the Thingiverse download for what you've been using then I'll see if I can replicate the issue with that model too.

Maybe it's an issue with multi-part STLs?
